master/usage/openstack/multiple-regions.md

```diff
@@ -0,0 +1,115 @@
+> from sending a kind of traffic) it can do that either as an IP prefix or as a
+> 'remote' security group ID, meaning the VMs that belong to that security
+> group.  But a 'remote' security group ID has to be in the same region as
+> where the overall security group is being defined, and can only identify VMs
+> within that same region.  {: .alert .alert-info}
+
+## Installation
+
+To install a multi-region OpenStack deployment with Calico, proceed
+region-by-region and follow [the normal procedure for adding Calico to an
+OpenStack
+region]({{site.baseurl}}/{{page.version}}/getting-started/openstack/installation),
+except for these points:
+
+1.  Only install a single etcd database (instead of one per region) and
+    configure all of the regions to use that as their {{site.prodname}}
+    datastore.
+
+1.  In `/etc/calico/felix.cfg` on each compute host, add
+
+    ```
+    [global]
+    OpenstackRegion = <region>
+    ```
+
+    where `<region>` is the name of the region that that compute host belongs to.
+
+1.  In `/etc/neutron/neutron.conf` on each controller and compute node, add
+
+    ```
+    [calico]
+    openstack_region = <region>
+    ```
+
+    where `<region>` is the name of the region that that node belongs to.
+
+## Configuring cross-region policy
+
+Suppose that:
+
+- you have two regions, "RegionOne" and "RegionTwo"
+
+- you have a set of VMs in RegionOne belonging to security group
+  a7734e61-b545-452d-a3cd-0189cbd9747a
+
+- you have a set of VMs in RegionTwo belonging to security group
+  85cc3048-abc3-43cc-89b3-377341426ac5
+
+- you want to allow the VMs in RegionTwo to connect to port 80 of the VMs in
+  RegionOne.
+
+You could do that by configuring this {{site.prodname}} policy:
+
+```yaml
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: allow-tcp-80
+  namespace: openstack-region-RegionOne
+spec:
+  selector: "has(sg.projectcalico.org/openstack-a7734e61-b545-452d-a3cd-0189cbd9747a)"
+  types:
+  - Ingress
+  ingress:
+  - action: Allow
+    protocol: TCP
+    source:
+      namespaceSelector: "all()"
+      selector: "has(sg.projectcalico.org/openstack-85cc3048-abc3-43cc-89b3-377341426ac5)"
+    destination:
+      ports:
+      - 80
+```
+
+In words, this says that connections to port 80 of the VMs with the label for
+the a7734e61... security group are allowed from VMs - in any namespace - with
+the label for the 85cc3048... security group.  We know that the 85cc3048... VMs
+_are_ in a different namespace from the a7734e61... VMs, so the
+`namespaceSelector: "all()"` here - which allows the connecting VMs to come
+from any region or namespace - is critical for this policy to work as intended.
+
+You can use any of the
+[labels]({{site.baseurl}}/{{page.version}}/usage/openstack/labels) that
+{{site.prodname}} adds to OpenStack VM endpoints, to identify a set of allowed
+(or denied) clients.  You can also use `nets` or `notNets` instead of
+`selector`, to identify clients by IP address.
```

Comment:
Example looks good but could we give them some more examples of useful selectors and maybe a GlobalNetworkPolicy example?  I'm worried that people will have difficulty in coming up with their own.  For example:

```
Here are some more examples of selectors to get you started:

- To select all VMs in a RegionOne: `projectcalico.org/namespace == 'openstack-RegionOne'`
- To select all VMs that are in a security group named "production" in *either* region: `has(sg-name.projectcalico.org/openstack-production)`

```

Reply:
Thanks, I've switched to Global and worked that text in.